### DISCUSSION TITLE
How can I get vertico to choose template text as the first selection?

### DISCUSSION THREAD

**Discussion by Bujiraso** [minad/vertico#479] [Q&A] [answered]
## Background
Consider the following ELISP
```
(find-file (read-file-name "Pick a file: " "~/" nil nil "sample.txt"))
```
On a bare `emacs -Q` config, by default, when you run the following ELISP, you get a selection of `~/sample.txt`. Hitting enter opens that file. This is the desired functionality.

Pleasantly, [selectrum](https://github.com/radian-software/selectrum/) maintains this functionality, if you install that.

## Problem

If you install Vertico, you get something somewhat visually nicer than Selectrum for the case above, but pragmatically perhaps more troublesome. 

* On the good side, you get to see all kinds of options, enhancing the "template" aspect of this argument; Emacs and Selectrum don't do this.
* On the "I would really like to change this" side, if you have any files in the folder (I picked `~` so that you hopefully do!), you will unfortunately have selected the top option for an already existing file, not the suggested `sample.txt`. It's a bit of a bait and switch, visually, and also requires more key strokes to get to the same result as the default or Selectrum mode selection would.

## Question

How can I make vertico behave like the default Emacs (and Selectrum's) autocomplete? I want this `sample.txt` to be already selected, as it's filled out and all.

I'm happy to accept "work-around"s but would love to see a clean solution, even if a code change or PR would be necessary.
I really like vertico so far, but this has been interrupting my flow a lot and now I'm side-eyeing dropping back to Selectrum, but I really don't want to, so please do help if you can!

### Non-Solutions
`vertico-preselect` does not seem to impact this. I tried all four options.

**Comment by minad** [comment]
I get the desired functionality you described. This likely depends on your settings of the completion-styles.

**Reply by Bujiraso** [reply]
Could you share yours? I'd like to try it

**Reply by minad** [reply]
See the README, I mostly use the documented configurations.

**Comment by Bujiraso** [comment, marked as answer]
This behavior occurs due to `emacs22` being in my `completion-styles` list. Removing it "fixes" the functionality since, as [this page](https://www.gnu.org/software/emacs/manual/html_node/emacs/Completion-Styles.html) notes, this behavior that i don't like is actually seemingly part of its design.

Quote: `it ignores the text in the minibuffer after point.`
